anchor/common/qbft/src/lib.rs

```
@@ -1108,18 +1171,29 @@ where
         &self,
         msg_type: QbftMessageType,
         data_hash: D::Hash,
-        mut round_change_justification: Vec<SignedSSVMessage>,
-        mut prepare_justification: Vec<SignedSSVMessage>,
-    ) -> UnsignedWrappedQbftMessage {
+        round_change_justification: Vec<SignedSSVMessage>,
+        prepare_justification: Vec<SignedSSVMessage>,
+    ) -> Option<UnsignedWrappedQbftMessage> {
         let data = self.get_message_data(&msg_type, data_hash);
 
         // Clear full_data from justifications as these do not store full data.
-        for round_change_justification in &mut round_change_justification {
-            round_change_justification.set_full_data(vec![]);
-        }
-        for prepare_justification in &mut prepare_justification {
-            prepare_justification.set_full_data(vec![]);
-        }
+        let round_change_justification_vec: Vec<VariableList<u8, RoundChangeLength>> =
+            round_change_justification
+                .into_iter()
+                .map(|msg| msg.without_full_data())
+                .filter_map(|msg| to_variable_list(msg.as_ssz_bytes()))
```

error being ignored here?

---

fixed